app/proprietary/src/main/java/stirling/software/proprietary/security/controller/api/AuthController.java

```diff
@@ -0,0 +1,272 @@
+import stirling.software.proprietary.security.service.UserService;
+
+/** REST API Controller for authentication operations. */
+@RestController
+@RequestMapping("/api/v1/auth")
+@RequiredArgsConstructor
+@Slf4j
+@UserApi
+public class AuthController {
+
+    private final UserService userService;
+    private final JwtServiceInterface jwtService;
+    private final CustomUserDetailsService userDetailsService;
+
+    /**
+     * Login endpoint - replaces Supabase signInWithPassword
+     *
+     * @param request Login credentials (email/username and password)
+     * @param response HTTP response to set JWT cookie
+     * @return User and session information
+     */
+    @PostMapping("/login")
+    public ResponseEntity<?> login(
+            @RequestBody LoginRequest request, HttpServletResponse response) {
+        try {
+            log.debug("Login attempt for user: {}", request.email());
+
+            UserDetails userDetails = userDetailsService.loadUserByUsername(request.email());
+            User user = (User) userDetails;
+
+            if (!userService.isPasswordCorrect(user, request.password())) {
+                log.warn("Invalid password for user: {}", request.email());
+                return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
+                        .body(Map.of("error", "Invalid credentials"));
+            }
+
+            if (!user.isEnabled()) {
+                log.warn("Disabled user attempted login: {}", request.email());
+                return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
+                        .body(Map.of("error", "User account is disabled"));
+            }
+
+            Map<String, Object> claims = new HashMap<>();
+            claims.put("authType", AuthenticationType.WEB.toString());
+            claims.put("role", user.getRolesAsString());
+
+            String token = jwtService.generateToken(user.getUsername(), claims);
+
+            log.info("Login successful for user: {}", request.email());
+
+            return ResponseEntity.ok(
+                    Map.of(
+                            "user", buildUserResponse(user),
+                            "session", Map.of("access_token", token, "expires_in", 3600)));
+
+        } catch (AuthenticationException e) {
+            log.error("Authentication failed for user: {}", request.email(), e);
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
+                    .body(Map.of("error", "Invalid credentials"));
+        } catch (Exception e) {
+            log.error("Login error for user: {}", request.email(), e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                    .body(Map.of("error", "Internal server error"));
+        }
+    }
+
+    /**
+     * Registration endpoint
+     *
+     * @param request Registration details (email, password, name)
+     * @return User information or error
+     */
+    @PostMapping("/register")
```

Comment:
no guarded in any way, this allows any user registeration right? Also Auth vs user create, seems like a weird mismatch here we already have endpoints for user creation for admins etc

If the goal is to allow ANY user to register an account, this should be config and disabled at start as well but agin, scope change, V1 didnt do this